modules/core/src/main/scala/com.fortysevendeg.hood/benchmark/BenchmarkService.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.fortysevendeg.hood.benchmark
+
+import cats.effect.Sync
+import com.fortysevendeg.hood.model.Benchmark
+
+sealed trait BenchmarkComparisonStatus
+case object OK      extends BenchmarkComparisonStatus
+case object Warning extends BenchmarkComparisonStatus
+case object Error   extends BenchmarkComparisonStatus
+
+case class BenchmarkComparisonResult(
+    previous: Benchmark,
+    current: Benchmark,
+    result: BenchmarkComparisonStatus)
+
+trait BenchmarkService[F[_]] {
+
+  def compare(
+      currentBenchmark: Benchmark,
+      previousBenchmark: Benchmark,
+      threshold: Double): F[BenchmarkComparisonResult]
+
+}
+
+object BenchmarkService {
+
+  def build[F[_]: Sync]: BenchmarkService[F] = new BenchmarkServiceImpl[F]
+
+  class BenchmarkServiceImpl[F[_]](implicit S: Sync[F]) extends BenchmarkService[F] {
+
+    def compare(
+        currentBenchmark: Benchmark,
+        previousBenchmark: Benchmark,
+        threshold: Double): F[BenchmarkComparisonResult] = S.delay {
```

Comment:
There is any way this could be unpure? I think we don't need the `S.delay` here and can just return `BenchmarkComparisonResult`